app/lib/return-cycle-dates.lib.js

```diff
@@ -0,0 +1,207 @@
+'use strict'
+
+/**
+ * General helper methods
+ * @module DatesLib
+ */
+
+const { returnCycleDates } = require('./static-lookups.lib.js')
+const { formatDateObjectToISO } = require('./dates.lib.js')
+
+/**
+ * Get the due date of next provided cycle, either summer or winter and all year, formatted as YYYY-MM-DD
+ *
+ * @param {boolean} summer - true for summer, false for winter and all year.
+ * @returns {string} - the due date of the next cycle as an ISO string.
+ */
+function cycleDueDateAsISO (summer) {
+  return formatDateObjectToISO(cycleDueDate(summer))
+}
+
+/**
+ * Get the due date of next provided cycle, either summer or winter and all year
+ *
+ * @param {boolean} summer - true for summer, false for winter and all year.
+ * @returns {Date} - the due date of the next cycle.
+ */
+function cycleDueDate (summer) {
+  const today = new Date()
+  const year = today.getFullYear()
+  const month = today.getMonth()
+
+  if (summer) {
+    if (month > returnCycleDates.summer.endDate.month) {
+      return new Date(year + 1, returnCycleDates.summer.dueDate.month, returnCycleDates.summer.dueDate.day)
+    }
+
+    return new Date(year, returnCycleDates.summer.dueDate.month, returnCycleDates.summer.dueDate.day)
+  }
+
+  if (month > returnCycleDates.allYear.endDate.month) {
+    return new Date(year + 1, returnCycleDates.allYear.dueDate.month, returnCycleDates.allYear.dueDate.day)
+  }
+
+  return new Date(year, returnCycleDates.allYear.dueDate.month, returnCycleDates.allYear.dueDate.day)
+}
+
+/**
+ * Given an arbitary date and if it is summer or all-year return the due date of that cycle
+ *
+ * @param {Date} date - the date whose due date you want to find.
+ * @param {boolean} summer - true for summer, false for winter and all year.
+ * @returns {string} - the due date of the next cycle.
+ */
+function cycleDueDateByDate (date, summer) {
+  const year = date.getFullYear()
+  const month = date.getMonth()
+
+  if (summer) {
+    if (month > returnCycleDates.summer.endDate.month) {
+      return formatDateObjectToISO(new Date(`${year + 1}-${returnCycleDates.summer.dueDate.month + 1}-${returnCycleDates.summer.dueDate.day}`))
+    }
+
+    return formatDateObjectToISO(new Date(`${year}-${returnCycleDates.summer.dueDate.month + 1}-${returnCycleDates.summer.dueDate.day}`))
+  }
+
+  if (month > returnCycleDates.allYear.endDate.month) {
+    return formatDateObjectToISO(new Date(`${year + 1}-${returnCycleDates.allYear.dueDate.month + 1}-${returnCycleDates.allYear.dueDate.day}`))
+  }
+
+  return formatDateObjectToISO(new Date(`${year}-${returnCycleDates.allYear.dueDate.month + 1}-${returnCycleDates.allYear.dueDate.day}`))
+}
```

Comment:
This applies to a few of the methods in this file. Our linter is not picking this up but some of the lines have gone beyond 120 chars. I think we can resolve this and make things a bit clearer at the same time. My suggestion would be the following.

```suggestion
function cycleDueDateByDate (date, summer) {
  const year = date.getFullYear()
  const month = date.getMonth()

  let cycleDueYear
  let cycleDueMonth
  let cycleDueDay

  if (summer) {
    cycleDueDay = returnCycleDates.summer.dueDate.day
    cycleDueMonth = returnCycleDates.summer.dueDate.month + 1

    cycleDueYear = month > returnCycleDates.summer.endDate.month ? year + 1 : year
  } else {
    cycleDueDay = returnCycleDates.allYear.dueDate.day
    cycleDueMonth = returnCycleDates.allYear.dueDate.month + 1

    cycleDueYear = month > returnCycleDates.allYear.endDate.month ? year + 1 : year
  }

  const cycleDueDate = new Date(`${cycleDueYear}-${cycleDueMonth}-${cycleDueDay}`)

  return formatDateObjectToISO(cycleDueDate)
}
```

But open to other ideas. I think the current pattern is based on my harping on about the 'guard clause' pattern. But I think this is a good example of an exception to that pattern.